lib/nm/index.ts

```diff
@@ -0,0 +1,121 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import NetworkManagerTypes from './types';
+
+const dbus: any = require('dbus-native');
+
+const systemBus: any = dbus.systemBus();
+const service: string[] = [ 'org', 'freedesktop', 'NetworkManager' ];
+
+export interface WifiDevice {
+	devicePath: string;
+	type: number;
+}
+
+export interface AccessPoint {
+	apPath: string;
+	apSsid: string;
+}
+
+export class NetworkManager extends NetworkManagerTypes {
+	wifiDevice: WifiDevice;
+	accessPoints: AccessPoint[];
+	wifiActive: boolean;
+
+	constructor() {
+		super();
+
+		this.wifiActive = true;
+
+		this.getWifiDevice()
+		.then((device: WifiDevice) => {
+			const service = Promise.promisifyAll(systemBus.getService('org.freedesktop.NetworkManager'));
+			service.getInterfaceAsync(device.devicePath, 'org.freedesktop.NetworkManager.Device.Wireless')
+			.then((iface) => {
+				console.log('YOLO');
+				iface.on('AccessPointAdded', (...data) => {
+					console.log('YAY ACCESS POINT ADDED', ...data);
+				});
+				iface.on('AccessPointRemoved', (...data) => {
+					console.log('YAY ACCESS POINT REMOVED', ...data);
+				});
+				console.log('SBOLO');
+			})
+			.catch((err) => {
+				console.error('gnappo', err);
+			});
+		});
+	}
```

Comment:
You could use `async` / `await`, that would make things shorter and more readable.
The async part shouldn't be in the constructor IMO, otherwise we'll never be able to catch its exceptions. I'd add an `async init()` method.

Reply:
The reason I didn't use `async`/`await` is for what you suggested in another comment: using Bluebird with the name `Promise`, I got conflicts when trying to use async/await because of course, it reserves the word.
Having changed `Promise` to `Bluebird`, I'm gonna do a short refactor, at least providing functions with correct names for each async operation, be it with or without async/await. I'll wait for other opinions on including it, since I think it needs at least `node v7.6+` in order to be running and I'm not sure of the future architecture of the container holding this service

Reply:
With typescript async/await works with node6. `async` functions get transpiled to generators if you target `es6` in your `tsconfig.json`. It can even transpile it for es3, see https://blog.mariusschulz.com/2016/12/09/typescript-2-1-async-await-for-es3-es5

Reply:
Oh nice! Then I'm gonna include it as soon as I'm done with toggle/general refactor, which will be much easier anyway to translate to `async`/`await`